common/backoff/cron.go

```diff
@@ -68,9 +77,13 @@ func GetBackoffForNextSchedule(cronSchedule string, scheduledTime time.Time, now
 	} else {
 		nextScheduleTime = schedule.Next(scheduledUTCTime)
 		// Calculate the next schedule start time which is nearest to now (right after now).
-		for nextScheduleTime.Before(nowUTC) {
+		for !nextScheduleTime.IsZero() && nextScheduleTime.Before(nowUTC) {
 			nextScheduleTime = schedule.Next(nextScheduleTime)
 		}
+		if nextScheduleTime.IsZero() {
+			// no time can be found to satisfy the schedule
+			return InfiniteBackoff
+		}
```

Comment:
I think if we move this before the for loop, we can remove the check from the for conditional.

Reply:
I might be overlooking sth, but it looks like `NoBackoff` is used as a measure to prevent retries altogether. Do you think we should return that instead of a very long backoff?

Reply:
You are right. I can use the NoBackoff.